packages/ra-core/src/dataProvider/withLifecycleCallbacks.ts

```diff
@@ -123,6 +123,40 @@ export const withLifecycleCallbacks = (
     dataProvider: DataProvider,
     handlers: ResourceCallbacks[]
 ): DataProvider => {
+    /**
+     * Apply callbacks to the params for the given resource and hook
+     * @param {string} resource The resource name
+     * @param {string} hook The hook name (beforeGetList, afterGetOne, etc.)
+     * @param {U} params The params / result to pass to the callbacks
+     * @returns {Promise<U>} The params / result after the callbacks have been applied
+     */
+    const applyCallbacks = async function <U>(
```

Comment:
I suggest you extract that function, and unit test it.

Reply:
If I extract this function, I see two options :
1. either we have to pass the handlers + dataProvider each time, the drawback is that it's quite verbose and the code is less clear :
```javascript
newParams = await applyCallbacks(
                dataProvider,
                handlers,
                resource,
                'beforeGetList',
                newParams
            );
```
2. Or I could bind the the repeated parameters :
```typescript
    const applyCallbacksWithContext = applyCallbacks.bind(
        null,
        dataProvider,
        handlers
    );

....

newParams = await applyCallbacksWithContext(
                resource,
                'beforeGetList',
                newParams
            );

```

I tend to prefer the first option because the code is less complex to understand even though it's very verbose.

What is your opinion on that ?